stdlib/messaging/kafka/src/main/ballerina/src/kafka/producer.bal

```diff
@@ -126,28 +136,49 @@ public type SerializerType SER_BYTE_ARRAY|SER_STRING|SER_INT|SER_FLOAT;
 public type Producer client object {
 
     public ProducerConfig? producerConfig = ();
-    private string keySerializer;
-    private string valueSerializer;
+    private string keySerializerType;
+    private string valueSerializerType;
+    private Serializer? keySerializer = ();
+    private Serializer? valueSerializer = ();
 
     # Creates a new Kafka `Producer`.
     #
     # + config - Configurations related to initializing a Kafka `Producer`.
     public function __init(ProducerConfig config) {
         self.producerConfig = config;
-        self.keySerializer = config.keySerializer;
-        self.valueSerializer = config.valueSerializer;
-        var result = self.init(config);
-        if (result is error) {
-            panic result;
+        self.keySerializerType = config.keySerializerType;
+        self.valueSerializerType = config.valueSerializerType;
+
+        if (self.keySerializerType == SER_CUSTOM) {
+            var keySerializerObject = config.keySerializer;
+            if (keySerializerObject is ()) {
+                ProducerError e = error(PRODUCER_ERROR, message = "Invalid keySerializer config: Please Provide a " +
```

Comment:
This initialisation is unnecessary. Shall we remove it?

Reply:
Done.